.github/workflows/ci-core.yml

```diff
@@ -40,8 +40,12 @@ jobs:
           github.event_name == 'schedule' ||
           github.event_name == 'workflow_dispatch'
         }}
-    runs-on: ubuntu-latest
+    # running on macos allows longer list of files than linux
```

Comment:
This seems a little hacky. Does it have other effects? What would it take to work around the limitation?